backend/protos/xyz/block/ftl/v1/ftl.proto

```diff
@@ -375,3 +375,129 @@ service RunnerService {
   // Terminate the deployment on this Runner.
   rpc Terminate(TerminateRequest) returns (RegisterRunnerRequest);
 }
+
+message ConfigRef {
+  optional string module = 1;
+  string name = 2;
+}
+
+enum ConfigProvider {
+  // Write values inline in the configuration file.
+  CONFIG_INLINE = 0;
+
+  // Print configuration as environment variables.
+  CONFIG_ENVAR = 1;
+}
+
+message ListConfigRequest {
+  optional ConfigProvider provider = 1;
+  optional string module = 2;
+  optional bool include_values = 3;
+}
+message ListConfigResponse {
+  message Config {
+    string refPath = 1;
+    optional string value = 2;
+  }
+  repeated Config configs = 1;
+}
+
+message GetConfigRequest {
+  optional ConfigProvider provider = 1;
+  ConfigRef ref = 2;
+}
+message GetConfigResponse {
+  string value = 1;
+}
+
+message SetConfigRequest {
+  optional ConfigProvider provider = 1;
+  ConfigRef ref = 2;
+  string value = 3;
```

Comment:
Secret/config values are arbitrary JSON-encoded values, so this need to be `bytes`.